google/cloud/spanner/internal/time_format.cc

```diff
@@ -70,6 +169,38 @@ std::size_t ParseTime(char const* fmt, std::string const& s, std::tm* tm) {
 #endif
 }
 
+std::size_t ParseTime(std::string const& s, std::tm* tm) {
+  constexpr auto kYearMin = std::numeric_limits<int>::min() + 1900;
+  constexpr auto kYearMax = std::numeric_limits<int>::max();
+  std::tm tmp{};
+  char const* dp = ParseInt(s.c_str(), kYearMin, kYearMax, &tmp.tm_year);
+  if (dp != nullptr && *dp == '-') {
```

Comment:
> nit: I prefer to return early instead of indenting

Yeah ... a possibility.  But in this particular case I believe I find indenting more readable.  Certainly having only one "success" and one "error" `return` helps a little.